### PR TITLE
feat: allow specifying primary module kind

### PR DESCRIPTION
**Pull request by elsirion**
The primary module instance id cannot be known upfront, it may differ from federation to federation.

### CODE REVIEW

File: fedimint-client/src/lib.rs [hunk 1]
```diff
@@ -2358,6 +2361,10 @@ impl ClientBuilder {
     ///
     /// ## Panics
     /// If there was a primary module specified previously
+    #[deprecated(
+        since = "0.6.0",
+        note = "Use `with_primary_module_kind` instead, as the instance id can't be known upfront"
```
Comment (dpc) [inline fedimint-client/src/lib.rs:2366]: Being able to select a specific primary module by id is theoretically useful. It's just not something that typically should be done. Maybe we should rename and document better, but ultimately keep?

Comment (dpc) [reply fedimint-client/src/lib.rs:2366]: Deprecating probably doesn't help much as a separate step anyway.

**Comment by maan2003** [comment]
request to backport to 0.5

File: fedimint-client/src/lib.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-client/src/lib.rs:2366]: I'd say let's create a fn with a new name that retains the old behavior, but is strongly discouraged. I'd like to phase out the old one to get rid of old and likely incorrect usage.

Comment (dpc) [reply fedimint-client/src/lib.rs:2366]: Yes, we can just rugpull that name. :D

**Review by dpc — APPROVED** [review]

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.5`:
- #6534

**Pull request merged**